mautic/mautic
# [DPMMA-2341] Added Preference Center blocks to GrapesJS builder

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [❌ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #13544 ... 



#### Description:


PR adds Preference Center blocks to the GrapesJS builder. Related PR in grapesjs-preset-mautic: https://github.com/mautic/grapesjs-preset-mautic/pull/46 

![preference_center_blocks](https://github.com/mautic/mautic/assets/102536220/688e62b1-6d8c-4868-9a13-3390555e9d11)

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new landing page and switch "Set as preference center page" to Yes.
3. Open the GrapesJS builder and look for the Preference Center section in the "Open Blocks" panel on the right.
4. Drag the selected blocks onto the canvas. The preferences page must contain a "Save Preferences" button.
5. Save the page and attach it to an email. Check if the preference center is displayed correctly in the sent message.
